fix atxi pagination OoB possibility

whilei opened the pull request:
problem: atxi pagination param can cause slice OoB
solution: ensure param within len lim

Fixes #626

tzdybal commented on core/atxi.go:
```diff
@@ -367,9 +367,13 @@ func GetAddrTxs(db ethdb.Database, address common.Address, blockStartN uint64, b
+		} else if paginationStart > len(s) {
+			paginationStart = len(s)
 		}
 		if paginationEnd < 0 {
```
This can be simplified to:
```
if paginationEnd < 0 || paginationEnd > len(s) {
    paginationEnd = len(s)
}
```

whilei commented:
@tzdybal Fixed, and refactored `core/GetAtxi` method to return an error, if any. Also added an additional validation that `! paginationStart > paginationEnd`.

whilei commented:
Maybe actually fixed.

whilei commented:
Just a heads up that this failing test is #630, fixed with #632, and is unrelated to these changes.

tzdybal commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r2, 3 of 3 files at r3.
Review status: all files reviewed, 1 unresolved discussion (waiting on @tzdybal and @r8d8)

---



*Comments from [Reviewable](https://reviewable.io/reviews/ethereumproject/go-ethereum/627#-:-LF2NTLy7hV4PgSHZo0H:bnfp4nl)*
<!-- Sent from Reviewable.io -->
